src/main/java/org/codevillage/JavaEntityFactory.java

```diff
@@ -0,0 +1,21 @@
+package org.codevillage;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+
+public class JavaEntityFactory implements EntityFactory{
+
+    @Override
+    public JavaEntity createEntityFromFile(File filename) throws IOException {
+        TypeParsingStep type = new TypeParsingStep(null);
```

Comment:
This will be `EntityParsingChain p = new StepOne(new StepTwo(new Step3(new Step4(new ChainEnd))))` with the actual class names from the UML. We don't have all the steps merged in, but if you could build that out with what we have/make dummy classes for the others that would be awesome.